src/uu/stat/src/stat.rs

```diff
@@ -293,6 +318,93 @@ fn print_str(s: &str, flags: &Flags, width: usize, precision: Option<usize>) {
     pad_and_print(s, flags.left, width, Padding::Space);
 }
 
+fn quote_file_name(file_name: &str, quoting_style: &QuotingStyle) -> String {
+    match quoting_style {
+        QuotingStyle::Locale | QuotingStyle::Shell => {
+            let escaped = file_name.replace('\'', r"\'");
+            format!("'{}'", escaped)
+        }
+        QuotingStyle::Default => format!("\"{}\"", file_name),
+        QuotingStyle::Quote => file_name.to_string(),
+    }
+}
+
+fn get_quoted_file_name(
+    display_name: &str,
+    file: &OsString,
```

Comment:
Is `display_name` necessary? It looks to me like you could get it from `file` with `file.to_string_lossy()`.

Reply:
it seems legit but it isn't:
https://github.com/uutils/coreutils/blob/main/src/uu/stat/src/stat.rs#L639-L640
as you can see here, we are keeping the original name and changing :)